src/app/clusters/mode-base-server/mode-base-server.h

```diff
@@ -0,0 +1,215 @@
+    chip::Protocols::InteractionModel::Status UpdateOnMode(DataModel::Nullable<uint8_t> aNewOnMode);
+    chip::Protocols::InteractionModel::Status UpdateCurrentMode(uint8_t aNewMode);
+
+    // Attribute getters
+    DataModel::Nullable<uint8_t> GetStartUpMode() const;
+    DataModel::Nullable<uint8_t> GetOnMode() const;
+    uint8_t GetCurrentMode() const;
+    EndpointId GetEndpointId() const { return mEndpointId; }
+
+    /**
+     * Returns true if the feature is supported.
+     * @param feature the feature to check.
+     */
+    bool HasFeature(Feature feature) const;
+
+    /**
+     * This function returns true if the mode value given matches one of the supported modes, otherwise it returns false.
+     * @param mode
+     */
+    bool IsSupportedMode(uint8_t mode);
+
+private:
+    EndpointId mEndpointId{};
+    ClusterId mClusterId{};
+
+    // Attribute data store
+    uint8_t mCurrentMode;
+    DataModel::Nullable<uint8_t> mStartUpMode;
+    DataModel::Nullable<uint8_t> mOnMode;
+
+    uint32_t mFeature;
+
+    /**
+     * This checks to see if this clusters instance is a valid ModeBase aliased cluster based on the AliasedClusters list.
+     * @return true if the clusterId of this instance is a valid ModeBase cluster.
+     */
+    bool isAliasCluster() const;
+
+    /**
+     * Internal change-to-mode command handler function.
+     */
+    void handleChangeToMode(HandlerContext & ctx, const Commands::ChangeToMode::DecodableType & req);
+
+    /**
+     * Helper function that loads all the persistent attributes from the KVS. These attributes are CurrentMode,
+     * StartUpMode and OnMode.
+     */
+    void loadPersistentAttributes();
+
+public:
+    /**
+     * Creates a mode base cluster instance. The Init() function needs to be called for this instance to be registered and
+     * called by the interaction model at the appropriate times.
+     * @param aEndpointId The endpoint on which this cluster exists. This must match the zap configuration.
+     * @param aClusterId The ID of the ModeBase aliased cluster to be instantiated.
+     * @param aFeature The bitmask value that identifies which features are supported by this instance.
+     */
+    Instance(EndpointId aEndpointId, ClusterId aClusterId, uint32_t aFeature) :
+        CommandHandlerInterface(Optional<EndpointId>(aEndpointId), aClusterId),
+        AttributeAccessInterface(Optional<EndpointId>(aEndpointId), aClusterId), mEndpointId(aEndpointId), mClusterId(aClusterId),
+        mFeature(aFeature)
+    {
+        mCurrentMode = 0; // This is a temporary value and may not be valid. We will change this to the value of the first
+                          // mode in the list at the start of the Init function to ensure that it represents a valid mode.
+        mStartUpMode = DataModel::Nullable<uint8_t>(); // Initialised to null
+        mOnMode      = DataModel::Nullable<uint8_t>(); // Initialised to null
```

Comment:
Nullables start off initialized to null, so no need to manually do it.  For mCurrentMode, why is the init code in the ctor instead of a field initializer?

Reply:
Fixed.